File: Irrgarten_Ruby/Controller/controller.rb
require_relative '../game'
require_relative '../UI/textUI'
require_relative '../directions'

# Este módulo contiene el controlador del juego que irá mostrando
# la información del juego, así como actualizándolo
#
# @author Joaquin Avilés de la Fuente
# @author Arturo Olivares Martos
module Control

    # Esta clase se encarga mantener el control general del juego, es decir, su visualización
    # el estado del juego, su finalización, etc.
    #
    # @author Joaquin Avilés de la Fuente
    # @author Arturo Olivares Martos
    class Controller

      # Constructor del controlador
      #
      # @param game [Game] juego
      # @param view [TextUI] visualización
      def initialize(game,view)
        @game = game
        @view = view
      end
      
      # Juego
      def play
        end_of_game = false
        while (!end_of_game)
          @view.show_game(@game.game_state) 
          direction = @view.next_move 
          end_of_game = @game.next_step(direction)
        end
        @view.show_game(@game.game_state)
      end
    end # class   
  end # module        

File: Irrgarten_Java/src/irrgarten/controller/Controller.java
package irrgarten.controller;

import irrgarten.Directions;
import irrgarten.Game;
import irrgarten.UI.*;

/**
 * Esta clase se encarga mantener el control general del juego, es decir, su visualización
 * el estado del juego, su finalización, etc.
 * 
 * @author Arturo Olivares Martos
 * @author Joaquín Avilés de la Fuente
 */
public class Controller {
    
    /**
     * Juego a jugar
     */
    private Game game;
    
    /**
     * Vista del juego
     */
    private UI view;
    
    /**
     * Constructor del controlador
     * 
     * @param game Juego
     * @param view Clase que se encarga de la interacción con el usuario
     */
    public Controller(Game game, UI view) {
        this.game = game;
        this.view = view;
    }
    // TODO: UI se puede manejar así?
    
    /**
     * Método que se encarga de jugar al juego
     */
    public void play() {
        boolean endOfGame = false;
        while (!endOfGame) {
            view.showGame(game.getGameState());
            
            // TODO: Como parar y que sea cuando lo indica el usuario
            Directions direction = view.nextMove(); 
            endOfGame = game.nextStep(direction);
        }
      view.showGame(game.getGameState());        
    }
    
}